ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -88,9 +88,9 @@ def __init__(self, obs, reward, done, max_step, agent_id, action_mask):
         self.reward: np.ndarray = reward
         self.done: np.ndarray = done
         self.max_step: np.ndarray = max_step
-        self.agent_id: np.ndarray = agent_id
```

Comment:
I would rather keep the np.ndarray. It is for performance and compatibility with future communication.

Reply:
Converted it back as we discussed. It's actually not too bad since each add_experience call only needs a single int for worker id (the whole BatchedStepResult is from one worker, after all).